src/Build/BuildCheck/Infrastructure/BuildCheckManagerProvider.cs

```diff
@@ -539,10 +539,7 @@ public void ProcessProjectEvaluationStarted(
             string projectFullPath)
         {
             _projectsByEvaluationId[checkContext.BuildEventContext.EvaluationId] = projectFullPath;
-            if (!_deferredProjectEvalIdToImportedProjects.ContainsKey(checkContext.BuildEventContext.EvaluationId))
-            {
-                _deferredProjectEvalIdToImportedProjects.Add(checkContext.BuildEventContext.EvaluationId, [projectFullPath]);
-            }
```

Comment:
I am to blame for poorly commenting those events. This specific handler can be called as handler of EventArgs, but as well in worker node.
That's needed for the `_projectsByEvaluationId` structure that it was filling before.

The `_deferredProjectEvalIdToImportedProjects` feels as something that should be fed only from EventArgs. So technically this should be moved to a separate method that is called from `BuildCheckBuildEventHandler.HandleProjectEvaluationStartedEvent`. Thoughts @YuliiaKovalova ?

Reply:
tracking with https://github.com/dotnet/msbuild/issues/10960 so I'm going to push this now.